### PR TITLE
Fix output units: mV/nA → V/nA

### LINKED ISSUE #93: Clarify and fix output units across all electrode types
## Problem

The SONATA electrodes_file spec requires scaling_factors in **mV/nA**. Currently, different electrode types may produce inconsistent units:

- **PointSource / LineSource**: formula is `1/(4πσr)` with r in meters and a `× 1e-9` factor. Output is effectively V/nA.
- **Reciprocity / DipoleReciprocity**: formula is `potential / CurrentApplied`. The Sim4Life field files use Dirichlet boundary conditions (1V / -1V) and store the resulting current as `CurrentApplied` (e.g. 415527 in the test file `Infinite_Close_HighRes_SmallSphere.h5`). Dimensional analysis suggests output is V/A, which is `1e-6 mV/nA`. However, empirical comparison with point/line source at the same position shows only ~18× difference, not 1e6×.
- **ObjectiveCSD_***: dimensionless binary masks (0/1).

## Key questions

1. What units does Sim4Life export? The potential field range is [-1, 1] (normalized?), mesh axes are in meters, and `CurrentApplied = 415527` is clearly not 1 A. The paper (S1 Methods) describes Dirichlet BCs of ±1V, with current computed by surface integration of j = σE.
2. What conversion factor is needed on the reciprocity output to produce mV/nA?
3. Should point/line source use NEURON units (µm + S/m → mV/nA directly, no conversion) or SI with explicit conversion?

## Affected tests

- `tests/integration/test_weights.py::test_single_cell_write_weights_near` (uses Reciprocity + DipoleReciprocity + PointSource + LineSource)
- `tests/integration/test_weights.py::test_single_cell_write_weights_distant` (same mix)
- `tests/unit/test_physics.py::test_get_coeffs_reciprocity`
- `tests/unit/test_physics.py::test_get_coeffs_dipole_reciprocity`
- `tests/unit/test_physics.py::test_coefficients_are_in_mV_per_nA`

## Reference files

All `.h5` reference files under `examples/*/reference/` will need regeneration once units are fixed. The current references were generated by the old code and their unit semantics are inherited from that.

## Proposal

1. Adopt NEURON units internally: µm for distances, S/m for sigma, mV/nA for output
2. Point/line source: `1/(4πσr_µm)` gives mV/nA directly — no conversion factors
3. Reciprocity: determine correct factor from Sim4Life conventions (contact Joseph Tharayil)
4. Regenerate all reference files
5. Add explicit unit test asserting output is in mV/nA

### PR DESCRIPTION
**Pull request by cattabiani**
Point/line source output V/nA (SONATA spec) instead of mV/nA.

- Add 1e-3 factor with unit derivation comments
- Reciprocity already outputs V/nA (unchanged)
- Update references and unit tests

Fix: https://github.com/openbraininstitute/BlueRecording/issues/93

### CODE REVIEW

File: bluerecording/physics.py [hunk 1]
```diff
@@ -220,24 +223,30 @@ def _point_source_coeffs(
...
         positions_um: Segment positions (µm), shape ``(N_points, 3)``.
         electrode_positions_um: Electrode positions (µm), shape ``(N_elec, 3)``.
         sigma: Extracellular conductivity (S/m). Scalar or array of shape ``(N_elec,)``.
 
     Returns:
-        Coefficients array (mV/nA), shape ``(N_elec, N_points)``.
+        Coefficients array (V/nA), shape ``(N_elec, N_points)``.
     """
+    # Unit derivation for point source: V = I / (4π σ r)
+    # With σ in S/m and r in µm (= 1e-6 m):
```
Comment (mgeplf) [inline bluerecording/physics.py:237]: is this LLM output? ` (= 1e-6 m)` doesn't seem particularly useful to me, or things like `has units m/(S·m) = 1/S = Ω`

**Review by mgeplf — COMMENTED** [review]
I'm still confused as to how this worked in the past if the units were wrong; is there a another PR on the consuming side?

File: bluerecording/physics.py (hunk 1 above)
Comment (cattabiani) [reply bluerecording/physics.py:237]: no. The LLM actually did not want to add the units. I wanted it for future reference

**Comment by cattabiani** [comment]
@mgeplf 
>I'm still confused as to how this worked in the past if the units were wrong; is there a another PR on the consuming side?

previously it was secretly `V/nA` even if the spec was `mV/nA`. This is to go back to the past and remove as many magic numbers as possible

**Comment by cattabiani** [comment]
@mgeplf I created the problem based on the wrong spec here:

https://github.com/openbraininstitute/BlueRecording/pull/94

**Comment by mgeplf** [comment]
>  based on the wrong spec here:

ahhhh, ok, I finally see.

File: bluerecording/physics.py (hunk 1 above)
Comment (mgeplf) [reply bluerecording/physics.py:237]: huh, ok, thanks.

**Review by mgeplf — DISMISSED** [review]
